### DISCUSSION TITLE
How to get second FirebaseCrashlytics.instance by name?

### DISCUSSION THREAD

**Discussion by ndnam198** [firebase/flutterfire#11183] [Q&A]
#flutter #dart #firebase_crashlytics #multipleinstances

I am using FirebaseCrashlytics in a Flutter app. I am in need of create two Firebase instance, one for native Android with 
`"[DEFAULT]"` name, and the other customized with name `"FlutterFirebaseInstance"`

My setup: 

```yaml
firebase_core: ^2.14.0  
firebase_crashlytics: ^3.3.3
```

Implementation in Dart as follows: 

```Dart
  const String instanceName = 'FlutterFirebaseInstance';

  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(
    name: instanceName,
    options: DefaultFirebaseOptions.currentPlatform, // taking from firebase_options.dart
  );
```

then verify by printing apps via Firebase.app

```dart
  print('all firebase apps: ${Firebase.app(instanceName)}');
// ![image](https://github.com/firebase/flutterfire/assets/49189269/c249243e-459c-4fc6-8aff-d130d1ec56c2)
```

But I got no idea how to get FirebaseCrashlytics of this custom instance (my instance). Basically `FirebaseCrashlytics.instance` return the instance of name `"[DEFAULT]"` . So whenever to call `FirebaseCrashlytics.instance.recordError(...)`, the report goes to the host application rather than what is defined in my `firebase_options.dart` file.

```dart 
FirebaseCrashlytics.instance.recordError(...)
```

I checked `FirebaseCrashlytics` source code, but there's literally no way. Is it a bug or a soon-to-be-implemented feature?
 
Does anyone have any idea on how to get the instance of the second `FirebaseCrashlytics`?

**Comment by aletandil** [comment]
@ndnam198 hi there, I have the same problem here. I'm using flavors in Flutter and I have 2 instances of Firebase app but I can't get the specific crashlytics instance for each app and it's not reporting error as well
